src/Identity/Core/src/DataProtectorTokenProvider.cs

```diff
@@ -110,38 +126,60 @@ public virtual async Task<bool> ValidateAsync(string purpose, string token, User
                     var expirationTime = creationTime + Options.TokenLifespan;
                     if (expirationTime < DateTimeOffset.UtcNow)
                     {
+                        Logger.InvalidExpirationTime();
                         return false;
                     }
 
                     var userId = reader.ReadString();
                     var actualUserId = await manager.GetUserIdAsync(user);
                     if (userId != actualUserId)
                     {
+                        Logger.UserIdsNotEquals();
                         return false;
                     }
+
                     var purp = reader.ReadString();
                     if (!string.Equals(purp, purpose))
                     {
+                        Logger.PurposeNotEquals();
```

Comment:
Can you pass in purp and purpose to the log and print that out in the log, that should help diagnose when the wrong token is used